### PR TITLE
Feature add check binfile tool

### PR DESCRIPTION
**Pull request by xavi-pinsach**

### CODE REVIEW

File: cli.js [hunk 1]
```diff
@@ -297,6 +298,13 @@ const commands = [
         alias: ["pkv"],
         options: "-verbose|v",
         action: plonkVerify
+    },
+    {
+        cmd: "file info [binary.file] [zkey|ptau|r1cs|wtns]",
```
Comment (phated) [inline cli.js:303]:
```suggestion
        cmd: "file info [binary.file] [zkey|ptau|r1cs|wtns]",
```
This cmd sting is confusing. I believe it's indicating that `.file` should be `zkey` or `ptau` or `r1cs` or `wtns` but the general CLI syntax is `[optional arg 1] [optional arg 2]`

Can we reword?

File: cli.js [hunk 2]
```diff
@@ -297,6 +298,13 @@ const commands = [
...
+        description: "Check info of a binary file",
+        alias: ["fi"],
+        options: "-verbose|v",
```
Comment (phated) [inline cli.js:306]: Is this option used?

File: cli.js [hunk 3]
```diff
@@ -1091,3 +1099,53 @@ async function plonkVerify(params, options) {
...
+        sections.forEach((section, index) => {
+            let errors = [];
+            if (section.length > 1) errors.push(`Section ${index} has more than one section definition`);
+            else {
+                if (section[0].size === 0) {
+                    errors.push(`Section ${index} size is zero. This could generate side errors on other sections.`);
+                }
+            }
+            if(section[0].p + section[0].size > fd.totalSize) {
+                errors.push(`Section ${index} is out of bounds of he file.`);
```
Comment (phated) [inline cli.js:1135]:
```suggestion
                errors.push(`Section ${index} is out of bounds of the file.`);
```

File: cli.js [hunk 4]
```diff
@@ -1091,3 +1099,53 @@ async function plonkVerify(params, options) {
...
+        console.log(`Version:         ${fd.version}`);
+        console.log(`Bin version:     ${fd.binVersion}`);
+        console.log("");
+
+        sections.forEach((section, index) => {
+            let errors = [];
+            if (section.length > 1) errors.push(`Section ${index} has more than one section definition`);
+            else {
+                if (section[0].size === 0) {
+                    errors.push(`Section ${index} size is zero. This could generate side errors on other sections.`);
```
Comment (phated) [inline cli.js:1131]: What is a "side error"?

**Review by phated — COMMENTED** [review]
Love this feature! I had a few comments and questions

File: cli.js (hunk 1 above)
Comment (xavi-pinsach) [reply cli.js:303]: My mistake, in fact command string is cmd: "file info [binary.file]"

**Review by phated — APPROVED** [review]
Excellent! This looks great

**Pull request merged**